pyspeckit/cubes/SpectralCube.py

```diff
@@ -831,7 +835,12 @@ def fit_a_pixel(iixy):
             if errspec is not None:
                 sp.error = errspec
             elif errmap is not None:
-                sp.error = np.ones(sp.data.shape) * errmap[int(y),int(x)]
```

Comment:
Add a check here that `self.errorcube is None`; if it's not, raise an error and insist that it be one or the other.